src/hardware/input/mouseif_dos_driver.cpp

```diff
@@ -920,8 +889,120 @@ void MOUSEDOS_AfterNewVideoMode(const bool setmode)
 	state.update_region_y[1] = -1; // offscreen
 	state.cursor_type        = MouseCursor::Software;
 	state.enabled            = true;
+	state.inhibit_draw       = false;
 
-	clear_pending_events();
+	// Some software (like 'Down by the Laituri' game) is known to first set
+	// the min/max mouse cursor position and then set VESA mode, therefore
+	// (unless this is a driver reset) skip setting min/max position and
+	// granularity for SVGA graphic modes
+
+	if (is_mode_set && is_svga_mode && !is_svga_text) {
+		return;
+	}
+
+	// Helper lambda for setting text mode max position x/y
+
+	auto set_maxpos_text = [&]() {
+		constexpr uint16_t threshold_lo = 1;
+		constexpr uint16_t threshold_hi = 250;
+
+		constexpr uint16_t default_rows    = 25;
+		constexpr uint16_t default_columns = 80;
+
+		uint16_t columns = INT10_GetTextColumns();
+		uint16_t rows    = INT10_GetTextRows();
+
+		if (rows < threshold_lo || rows > threshold_hi) {
+			rows = default_rows;
+		}
+		if (columns < threshold_lo || columns > threshold_hi) {
+			columns = default_columns;
+		}
+
+		state.maxpos_x = static_cast<int16_t>(8 * columns - 1);
+		state.maxpos_y = static_cast<int16_t>(8 * rows - 1);
```

Comment:
Are these `8`'s related to "8-dot-mode text" vs "9-dot-mode text"?

Reply:
If so, the 8 vs. 9 dot mode sequence register bit is held in:

`vga.seq.clocking_mode.is_eight_dot_mode`, a new `bool VGA_GetEightDotMode()` function could be added to `vga.h` to get that state.

Reply:
I'll check. If so, also `move_cursor_seamless` might need to be adapted.